src/pykeen/nn/functional.py

```diff
@@ -956,6 +956,49 @@ def mure_interaction(
     ) + b_h + b_t
 
 
+def murp_interaction(
+    h: torch.FloatTensor,
+    b_h: torch.FloatTensor,
+    r_vec: torch.FloatTensor,
+    r_mat: torch.FloatTensor,
+    t: torch.FloatTensor,
+    b_t: torch.FloatTensor,
+) -> torch.FloatTensor:
+    """Evaluate the MuRP interaction function from [balazevic2019b]_.
+
+    :param h:
+    :param b_h:
+    :param r_vec:
+    :param r_mat:
+    :param t:
+    :param b_t:
+    :return:
+
+    The original code used the following idiom multiple times for many vectors (not only ``u``):
+
+    .. code-block:: python
+
+        u = torch.where(torch.norm(u, 2, dim=-1, keepdim=True) >= 1,
+                        u/(torch.norm(u, 2, dim=-1, keepdim=True)-1e-5), u)
+
+    The PyKEEN implementation already has :func:`pykeen.utils.clamp_norm`, which when given
+    ``maxnorm=1`` and ``eps=1e-5``, is effectively the same. This greatly allows for the
+    reduction of code verbosity in the following implementation.
+    """
+    h = clamp_norm(h, maxnorm=1)
+    t = clamp_norm(t, maxnorm=1)
+    r_mat = clamp_norm(r_mat, maxnorm=1)
```

Comment:
isn't `r_mat` a matrix? By default, the norm is computed along the last dimension only.

Reply:
yes, r_mat is a matrix. here's the code from the original implementation, which seems to be doing the same as the non-matrix ones:

https://github.com/ibalazevic/multirelational-poincare/blob/34523a61ca7867591fd645bfb0c0807246c08660/model.py#L30-L31

Reply:
`rvh` is the vector in the original code, i.e., `r_vec` for us; `Ru` is the matrix, i.e., `r_mat` for us.

Reply:
oops i have made a mistake then

Reply:
Fixed in b15914c0

Reply:
Now that I took a closer look at the paper and the code, both are vectors, since the matrix is modelled as a diagonal matrix.